apps/predbat/predbat.py

```diff
@@ -257,20 +290,29 @@ def __init__(self, base, id=0, quiet=False):
         else:
             self.soc_max = self.base.get_arg('soc_max', default=10.0, index=self.id) * self.base.battery_scaling
             self.nominal_capacity = self.soc_max
+
+            self.battery_voltage = 52.0
+            if('battery_voltage' in self.base.args):
+                self.base.get_arg('battery_voltage', index=self.id, default=52.0)   
+
             if self.inverter_type == 'GE':
                 self.battery_rate_max_raw = self.base.get_arg('charge_rate', attribute='max', index=self.id, default=2600.0)
             elif 'battery_rate_max' in self.base.args:
                 self.battery_rate_max_raw = self.base.get_arg('battery_rate_max', index=self.id, default=2600.0)
-            elif ('battery_charge_current_limit' in self.base.args) and ('battery_voltage' in self.base.args):
-                self.battery_rate_max_raw = self.base.get_arg('battery_charge_current_limit', index=self.id, default=65.0) * self.base.get_arg('battery_voltage', index=self.id, default=52.0)                 
+            elif 'battery_charge_current_limit' in self.base.args: 
+                    self.battery_rate_max_raw = self.base.get_arg('battery_charge_current_limit', index=self.id, default=65.0) * self.battery_voltage
+
+            else:
+                self.battery_rate_max_raw = 2600.0
+
             ivtime = self.base.get_arg('inverter_time', index=self.id, default=None)
 
         # Battery can not be zero size
         if self.soc_max <= 0:
             self.base.log("ERROR: Reported battery size from REST is {}, but it must be >0".format(self.soc_max))
             raise ValueError
 
-        # Battery rate max charge, discharge
+        # Battery rate max charge, discharge (all converted to kW/h)
```

Comment:
I think it's actually kWh/minute confusingly enough

Reply:
Yes, of course.